test/unit/linear_algebra/test_tridiagonal_solver.cpp

```diff
@@ -20,28 +25,40 @@ typedef std::vector<float> vecf;
 const double tol_dp = std::numeric_limits<double>::epsilon();
 const float tol_sp = std::numeric_limits<float>::epsilon();
 
+const std::size_t number_of_runs = 20;
+const bool diagonally_dominant = true;
+
+std::size_t sizes[5] = { 1000, 10000, 100000, 1000000, 10000000 };
```

Comment:
@AdityaDendukuri For unit test, `size = 1000` is probably enough for correctness check. Could you use `const std::size_t sizes = 1000;` instead (and revising the other code accordingly), and see if the memory check CI tests could pass for your PR (currently it seems hanging on the tridiagonal test).

Reply:
1 of the tests finished after 122 minutes. It passes, but that's waaaayyy too long for a CI test, I think